### PR TITLE
Add new automation support to runner

### PR DESCRIPTION
**Pull request by DrewScoggins**
We need to set the cli commit hash and commit date to the environment
before we run the benchmarks for the new reporting tool. I have taken
the code from where we do that as part of the sending to benchview and
put it right before we run the benchmarks. I also make sure that this is
reset after we run.

### CODE REVIEW

**Comment by DrewScoggins** [comment]
@billwert @adiaaida PTAL

File: scripts/benchmarks_ci.py [hunk 1]
```diff
@@ -237,13 +237,22 @@ def __main(args: list) -> int:
...
+            os.environ['PERFLAB_HASH'] = commit_sha
+            os.environ['PERFLAB_BUILDTIMESTAMP'] = source_timestamp
             micro_benchmarks.run(
                 BENCHMARKS_CSPROJ,
                 args.configuration,
                 framework,
                 verbose,
                 args
             )
+            os.environ['PERFLAB_HASH'] = ""
```
Comment (jorive) [inline scripts/benchmarks_ci.py]: I would just remove them from the environment instead of setting them to empty string.

Comment (jorive) [reply scripts/benchmarks_ci.py]: maybe use something like `os.unsetenv` or `os.environ.pop`?

**Comment by adiaaida** [comment]
And net461 (ie full framework runs) don't have this hash apparently?

**Comment by jorive** [comment]
Those parameters should be added as optional script parameters, and passed on automation (yml file).

**Comment by DrewScoggins** [comment]
Yeah, this currently will only work for our sdk runs, but since that is all we are doing right now I feel OK with that. I went ahead and made a change to make sure we don't break net461, but once the CI passed I am going to check this in.

**Pull request merged**